### DISCUSSION TITLE
appropriate citation for Flask module?

### DISCUSSION THREAD

**Discussion by mrneont** [pallets/flask#5390] [Q&A] [answered]
What is the appropriate citation for the Flask module, like in an academic paper? Preferably, in the form of:
AUTHOR[s] (year). title, etc.

**Comment by tachyondecay** [comment]
Please note that many citation formats do not require you to provide a formal citation for common software. For example, APA 7 guidance says:

> Common software and mobile apps mentioned in text, but not paraphrased or quoted, do not need citations…. If you used common software or mobile apps during your research, simply give the proper name of the software or app along with the version number in the text, if relevant.

Therefore, assuming your audience is familiar with Python, it is often sufficient to say, “This project used Flask 3.0,” etc.

If you need a formal citation, then in APA style it would be as follows:

> Pallets. (2024). _Flask_ (Version 3.0.1) [Computer software]. PyPI. https://palletsprojects.com/

The year should correspond to the release year of the version you used in your project. For example, if you used Flask 2.3, you would say 2023.

**Comment by mrneont** [comment, marked as answer]
OK, that works. I'm just trying to give credit where credit is due.  Many R modules, for example, have more standard looking citations.  I was wondering if there were something along that lines.  But this is fine, as well.  

FWIW, there isn't a single Flask version to point to in my case; users will have a wide range of version numbers on their computers depending on their package manager.
